scripts/zkai_gkr_d128_projection_scaling_preflight_gate.py

```
@@ -0,0 +1,669 @@
+def validate_source_numbers(sources: dict[str, dict[str, Any]]) -> None:
+    stwo_aggregate = require_dict(sources["stwo_gate"].get("aggregate"), "Stwo gate aggregate")
+    minimal_summary = require_dict(sources["minimal"].get("summary"), "minimal summary")
+    selector_summary = require_dict(sources["selector"].get("summary"), "selector summary")
+    shape_results = require_list(sources["shape"].get("results"), "shape results")
+    shape_sweep = require_list(sources["shape"].get("dimension_sweep"), "dimension sweep")
+
+    expect_equal(require_int(stwo_aggregate.get("baseline_local_typed_bytes"), "Stwo gate typed bytes"), EXPECTED_STWO_GATE_VALUE_TYPED_BYTES, "Stwo gate typed bytes")
+    expect_equal(require_int(stwo_aggregate.get("row_count"), "Stwo gate row count"), EXPECTED_D128_GATE_VALUE_ROWS, "Stwo gate row count")
+    expect_equal(require_int(minimal_summary.get("two_proof_frontier_typed_bytes"), "Stwo frontier typed bytes"), EXPECTED_STWO_FRONTIER_TYPED_BYTES, "Stwo frontier typed bytes")
+    expect_equal(require_int(minimal_summary.get("nanozk_reported_d128_block_proof_bytes"), "NANOZK reported bytes"), EXPECTED_NANOZK_REPORTED_BYTES, "NANOZK reported bytes")
+    expect_equal(require_bool(minimal_summary.get("missing_native_block_proof_object"), "missing native block flag"), True, "missing native block flag")
```

Ruff (0.15.12)

[warning] Boolean positional value in function call

(FBT003)